packages/components/grid/src/filter.ts

```diff
@@ -129,6 +132,21 @@ export class GridFilter<T = any> extends ScopedElementsMixin(LitElement) {
     super.disconnectedCallback();
   }
 
+  getFilterHeaderValue(): string {
+    let header = this.column.header;
+    if (typeof header !== 'string' && header !== undefined) {
+      const div = document.createElement('div');
+      div.innerHTML = (header as unknown as TemplateResult).strings[0];
+      const textNodes = Array.from(div.childNodes)
+        .filter(node => node.nodeType !== Node.ELEMENT_NODE && node.textContent?.trim())
+        .map(node => node.textContent?.trim());
+      header = textNodes.join(' ');
+    }
+    return (
```

Comment:
Can you change this so it returns early if there is a filter label? And do the "hard work" otherwise?